File: example/java/jimmer-sql-static/src/main/java/org/babyfish/jimmer/sql/example/dal/TreeNodeRepository.java
package org.babyfish.jimmer.sql.example.dal;

import org.babyfish.jimmer.Static;
import org.babyfish.jimmer.spring.repository.JRepository;
import org.babyfish.jimmer.sql.example.model.TreeNode;
import org.babyfish.jimmer.sql.fetcher.Fetcher;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface TreeNodeRepository extends JRepository<TreeNode, Long> {

    <S extends Static<TreeNode>> List<S> findByParentIsNullAndName(
            @Nullable String name,
            Class<S> staticType
    );
}

File: example/kotlin/jimmer-sql-static-kt/src/main/kotlin/org/babyfish/jimmer/sql/example/dal/TreeNodeRepository.kt
package org.babyfish.jimmer.sql.example.dal

import org.babyfish.jimmer.Static
import org.babyfish.jimmer.spring.repository.KRepository
import org.babyfish.jimmer.sql.example.model.TreeNode
import kotlin.reflect.KClass

interface TreeNodeRepository : KRepository<TreeNode, Long> {

    fun <S: Static<TreeNode>> findByParentIsNullAndName(
        name: String?,
        staticType: KClass<S>
    ): List<S>
}